src/include/kdbtypes.h

```diff
@@ -85,7 +89,7 @@ typedef unsigned char             kdb_octet_t;
 typedef signed short              kdb_short_t;
 typedef unsigned short            kdb_unsigned_short_t;
 
-#if SIZEOF_LONG == 4
+#if (LONG_MAX == +2147483647)
```

Comment:
@petermax2 SIZEOF_LONG is not defined
LONG_MAX is defined in ANSI C
http://www.open-std.org/jtc1/sc22/wg14/www/docs/n1256.pdf
So every compiler should support those numbers. lcms2.h and other projects have those values and they are used on so many platforms including OS X, Windows, Embedded ...